addons/beehave/nodes/composites/sequence_random.gd

```diff
@@ -0,0 +1,61 @@
+## This node will attempt to execute all of its children just like a 
+## [code]SequenceStar[/code] would, with the exception that the children 
+## will be executed in a random order.
+@tool
+class_name SequenceRandomComposite extends Composite
+@icon("../../icons/sequence_random.svg")
+
+
+## Whether the sequence should start where it left off after a previous failure.
+@export var resume_on_failure: bool = false
+## Whether the sequence should start where it left off after a previous interruption.
+@export var resume_on_interrupt: bool = false
+
+## A shuffled list of the children that will be executed in reverse order. 
+var _children_bag: Array[Node] = []
+var c: Node
+
+
```

Comment:
@lostptr could we provide an optional @export variable for the seed on both nodes? By default, it could be 'null' / not set and uses the current behaviour.

We will need that for predictable unit testing once we got the CI PR merged.

Reply:
It uses the global random number generator so you can set the seed anywhere in the game (preferably in a singleton for these kind of things). Adding an export variable allows for the seed to be set multiple times and change mid game which is not ideal. The seed should be set only once, thats why I suggesting the singleton.

Reply:
> The seed should be set only once, that's why I suggesting the singleton.

This is for the developer of the game to decide though, there are use-cases to set the seed individually. Not giving control over the randomness seems an unnecessary restriction.

Configuring the seed on a per-node basis is an additive change and can be done later. It will not collide with your proposed changes at all, so I am happy for it to be implemented later when there is truly a need for it!

Reply:
The worst thing is that we cant pass our own `RandomNumberGenerator` to the shuffle method, [it always uses the default global generator](https://github.com/godotengine/godot/blob/5674fa04aeb44a0d9cf48073fe2be25d1b804acf/core/variant/array.cpp#L618).